src/thunderbots/software/ai/passing/pass_generator.h

```diff
@@ -0,0 +1,229 @@
+#pragma once
+
+#include <thread>
+#include <mutex>
+
+#include "ai/passing/pass.h"
+#include "ai/world/world.h"
+#include "util/gradient_descent.h"
+#include "util/parameter/dynamic_parameters.h"
+#include "util/timestamp.h"
+
+namespace AI::Passing {
+
+    // TODO: detailed Javadoc comment for this class
+    // TODO: explain the general idiom of: get in, lock what you need, exit (RAII?)
+    // TODO: explain how to make changes to this class:
+    //      - if you add data, consider if it could be accessed by two threads at once. If so, it needs a mutex
+    //      - for each function added/changed, consider: what data does this access? If any of that data has a mutex, you must take a `lock_guard` for it
+    //      (NOTE: this is only for data accessed _directly_ by the function, any data accessed by functions called by the function should be in _those_ functions
+    class PassGenerator {
+
+    public:
+        // Delete the default constructor, we want to force users to choose what
+        // pass quality they deem reasonable
+        PassGenerator() = delete;
+
+        // Delete the copy and assignment operators because this class really shouldn't
+        // need them and we don't want to risk doing anything nasty with the internal
+        // threading this class uses
+        PassGenerator & operator=(const PassGenerator&) = delete;
+        PassGenerator(const PassGenerator&) = delete;
+
+        /**
+         * Create a PassGenerator with given parameters
+         *
+         * @param min_reasonable_pass_quality A value in [0,1] representing the minimum
```

Comment:
Should be extra explicit and specify that larger values are "better"

Reply:
yep, added